Warfare/Warfare/GameEngine.swift

```diff
@@ -17,6 +17,331 @@ class GameEngine {
         self.loadMap(number: "1")
     }
 
+    // MARK: - Operations
+
+
+    // Set up the gameState after which the player can start giving out orders
+    func beginTurn() {
+        for village in self.game.currentPlayer.villages {
+            for tile in village.controlledTiles {
+
+                // Replace tombstones
+                tile.replaceTombstone()
+
+                // Produce constructions and set unit actions.
+                // ReadyForOrders for all except first phase cultivation
+                if tile.unit?.currentAction == Constants.Unit.Action.StartCultivating {
+                    tile.unit?.currentAction = Constants.Unit.Action.FinishCultivating
+                } else {
+                    if tile.unit?.currentAction == Constants.Unit.Action.FinishCultivating {
+                        tile.land = .Meadow
+                    } else if tile.unit?.currentAction == Constants.Unit.Action.BuildingRoad {
+                        tile.structure = .Road
+                    }
+                    tile.unit?.currentAction = Constants.Unit.Action.ReadyForOrders
+                }
+
+                // Add gold value to village.
+                village.gold += tile.goldValue()
+
+                // Payout wages
+                village.gold -= tile.wage()
+            }
+
+            // Delete the Village
+            if village.gold <= 0 {
+                self.killVillage(village)
+            }
+        }
+    }
+
+    private func killVillage(village: Village) {
+        for tile in village.controlledTiles {
+            tile.structure = nil
+            if tile.unit != nil {
+                tile.unit = nil
+                tile.structure = .Tombstone
+            }
+            tile.village = nil
+            tile.owner.removeTile(tile)
+            self.game.neutralTiles.append(tile)
+        }
+        self.game.currentPlayer.removeVillage(village)
```

Comment:
doesn't removeVillage also call tile.clear for each tile controlled by the village?